python-lib/pagination.py

```diff
@@ -95,31 +122,41 @@ def update_next_page(self, data, response_links=None):
                 self.is_last_batch_empty = True
         else:
             batch_size = 1
+        if self.skip_key:
+            self.skip = data.get(self.skip_key)
+            logger.info("update_next_page_per_page:skip=data[{}]={}".format(self.skip_key, self.skip))
+        self.records_to_skip = self.records_to_skip + batch_size
```

Comment:
I'm not sure we need to keep track of `records_to_skip` for a page based method? AFAIK we only need to calculate `batch_size` to see if we are at the end?